### ISSUE TITLE
Incorrect emit of defining class expression with static property inside loop

### ISSUE THREAD

**Issue by yuit** [microsoft/TypeScript#8578]
As @jeffreymorlan  points out in the PR #8551, below code is emited incorrectly.

Compiler options: `--target es5`, `--target es6`

``` ts
const arr: {y(): number}[] = [];
for (let i = 0; i < 3; i++) {
    arr.push(class C {
        static x = i;
        static y = () => C.x * 2;
    });
}
arr.forEach(C => console.log(C.y()));
```

**Comment by RyanCavanaugh** [comment]
Verified

**Comment by jakebailey** [comment]
Just looking at old bugs; this one was fixed in #43197 and contains associated tests for class declarations in loops.
